middlewares/doctorAuthentication.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken'
+import CreateDoctorModel from "../models/doctormodel.js";
+
+const authenticateDoctor = async (req, res, next) => {
+ 
+
+  try {
+   
+    const decoded = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET);
```

Comment:
@robsdagreat I see on this line you  used the `token` variable, where did you define that variable? your codes will raise this error, so you have to fix this.
As a hint, the token should come from the request headers, so search about that and fix it

Reply:
Copy that sir, I thought that the headers are sent by the one who makes the login endpoint. I'm gong to change it asap